### PR TITLE
add spirv-tools/2020.3 + cmake components + pkg_config name + propagate C++ lib for C API

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **spirv-tools/2020.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I've introduced `tools.stdcpp_library` (conan 1.26.0) in this recipe https://github.com/conan-io/conan-center-index/issues/1918

File: recipes/spirv-tools/all/conanfile.py [hunk 1]
```diff
@@ -89,35 +96,61 @@ def package(self):
...
+        self.cpp_info.components["spirv-tools-reduce"].names["cmake_find_package_multi"] = "SPIRV-Tools-reduce"
+        self.cpp_info.components["spirv-tools-reduce"].libs = ["SPIRV-Tools-reduce"]
+        self.cpp_info.components["spirv-tools-reduce"].requires = ["spirv-tools-core", "spirv-tools-opt"]
 
         bin_path = os.path.join(self.package_folder, "bin")
         self.output.info('Appending PATH environment variable: %s' % bin_path)
         self.env_info.path.append(bin_path)
+
+    @property
+    def _stdcpp_library(self):
```
Comment (SSE4) [inline recipes/spirv-tools/all/conanfile.py:146]: I think we already added conan helper for that: https://docs.conan.io/en/latest/reference/tools.html?highlight=cppstd#tools-stdcpp-library

Comment (SpaceIm) [reply recipes/spirv-tools/all/conanfile.py:146]: Yes, but CI has not been updated yet: #1918

**Comment by SpaceIm** [comment]
By the way, how to raise ConanInvalidConfiguration if spirv-headers is overidden with non compatible version?

**Comment by uilianries** [comment]
@SpaceIm you can use `self.deps_cpp_info["spirv-headers"].version` under build method.

**Comment by SpaceIm** [comment]
@uilianries But build method is not called for pre-build binaries.

File: recipes/spirv-tools/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/spirv-tools/all/conanfile.py:146]: Now it's updated: https://github.com/conan-io/conan-center-index/wiki/Deployment-Changelog#24-june-2020---1055-cest

File: recipes/spirv-tools/all/conanfile.py [hunk 2]
```diff
@@ -36,15 +33,25 @@ def _source_subfolder(self):
     def _build_subfolder(self):
         return "build_subfolder"
 
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/spirv-tools/all/conanfile.py:40]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
```

**Comment by SpaceIm** [comment]
```
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppBuild.targets(312,5): error MSB3491: Could not write lines to file "x64\Debug\spirv-tools-spv-amd-shader-explicit-vertex-parameter\spirv-to.DA7493CC.tlog\spirv-tools-spv-amd-shader-explicit-vertex-parameter.lastbuildstate". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters. [C:\J\workspace\onan-center-pull-request_PR-1931\3\pr_1931_3_0_2\.short\9ec970\1\build_subfolder\source_subfolder\source\spirv-tools-spv-amd-shader-explicit-vertex-parameter.vcxproj]
```
This recipe has `short_paths=True`, so why this error?

File: recipes/spirv-tools/all/conandata.yml [hunk 3]
```diff
@@ -2,3 +2,6 @@ sources:
   "v2019.2":
     sha256: 1fde9d2a0df920a401441cd77253fc7b3b9ab0578eabda8caaaceaa6c7638440
     url: https://github.com/KhronosGroup/SPIRV-Tools/archive/v2019.2.tar.gz
+  "v2020.3":
```
Comment (Croydon) [inline recipes/spirv-tools/all/conandata.yml:5]: Probably to change now, but regularly we remove the `v` prefix if there is one

Comment (SpaceIm) [reply recipes/spirv-tools/all/conandata.yml:5]: It was discussed here when adding this recipe: https://github.com/conan-io/conan-center-index/pull/1349#issuecomment-627818533

**Comment by jgsogo** [comment]
I'm writing to the people in charge of the infrastructure. The path length limitation on Windows should have been removed, maybe this one is a rebel Windows node. I'll keep you updated.

**Comment by SpaceIm** [comment]
I should add a TODO. `SPIRV-Tools` exports one config file per target, and they aren't namespaced. It can't be modeled currently in conan for `cmake_find_package` and `cmake_find_package_multi` generators.

**Pull request merged**